substrate/primitives/runtime/src/lib.rs

```diff
@@ -401,6 +411,17 @@ impl traits::IdentifyAccount for MultiSigner {
 			Self::Ed25519(who) => <[u8; 32]>::from(who).into(),
 			Self::Sr25519(who) => <[u8; 32]>::from(who).into(),
 			Self::Ecdsa(who) => sp_io::hashing::blake2_256(who.as_ref()).into(),
+			Self::Eth(who) => {
+				// It is important that the account id is based off the eth address rather
+				// than its pubkey. This is because in many cases we don't know the pubkey
+				// of an eth account.
+				let eth_address = &sp_io::hashing::keccak_256(who.as_ref())[12..];
+				// This is by convention: `pallet_revive` maps eth addresses to account ids
+				// by filling up the additional 12 bytes with 0xEE.
+				let mut address = [0xEE; 32];
+				address[..20].copy_from_slice(eth_address);
+				address.into()
```

Comment:
could we expose a free fn here that we re-use in pallet-revive to keep it dry?

Reply:
This is a low level crate. We can't depend on a pallet as it would be a cyclic dependency. It would need to be the other way around.

The ideal way would be that the address mapping (including the state) is defined at the runtime level (or frame_system). And then it would parameterize the `MultiSignature`. This is because the mapping is basically a global concept and not only something confined to `pallet_revive`.

But in practice nobody will ever change it since it is the point that everybody uses the same mapping.